yt_dlp/extractor/youtube.py

```diff
@@ -2255,7 +2310,8 @@ def get_text(x):
                 default_client=ytm_client,
                 note='Downloading %sremix player API JSON' % ('mobile ' if force_mobile_client else ''))
 
-            ytm_streaming_data = try_get(ytm_player_response, lambda x: x['streamingData']) or {}
+            if isinstance(ytm_player_response, dict):
+                ytm_streaming_data = ytm_player_response.get('streamingData') or {}
```

Comment:
I hadn't thought about `ytm_player_response` not being a dict. In that case, you can just revert my changes and go back to:
```suggestion
           # ytm_player_response need not be a dict
            ytm_streaming_data = try_get(ytm_player_response, lambda x: x['streamingData']) or {}
```

but 
```py
            ytm_player_response = self._extract_response(...) or {}
            ytm_streaming_data = ytm_player_response.get('streamingData') or {}
```
is more clearer